Documentation/security/policy/standalone-dns-proxy.rst

```diff
@@ -0,0 +1,285 @@
+      standalone-dns-proxy-xxxxx    1/1     Running   0          1m
+
+7. **Apply DNS Policy**
+
+   Apply a policy that allows pods with the label ``org: alliance`` to query specific domains
+   (``cilium.io`` and its subdomains) and blocks all other queries:
+
+   .. literalinclude:: ../../../examples/policies/l7/dns/dns.yaml
+      :language: yaml
+
+   .. code-block:: shell-session
+
+      $ kubectl apply -f examples/policies/l7/dns/dns.yaml
+
+8. **Deploy Test Pod**
+
+   Create a test pod with the matching label:
+
+   .. code-block:: shell-session
+
+      $ kubectl run test-pod --image=nicolaka/netshoot --labels="org=alliance" -- sleep 3600
+
+9. **Verify DNS Policy Enforcement**
+
+   Test that DNS queries are being intercepted and resolved for allowed domains:
+
+   .. code-block:: shell-session
+
+      $ kubectl exec test-pod -- nslookup cilium.io.
+
+   Queries to ``cilium.io`` should succeed.
+
+   Now test that queries to non-allowed domains are refused:
+
+   .. code-block:: shell-session
+
+      $ kubectl exec test-pod -- nslookup example.com.
+      Server:         10.96.0.10
+      Address:        10.96.0.10#53
+
+      ** server can't find example.com: REFUSED
+
+   The query to ``example.com`` is refused because it's not in the allowed DNS policy rules.
+
+10. **Test Standalone Proxy Resilience**
+
+    Verify that the standalone DNS proxy continues to work even when the Cilium agent is down:
+
+    .. code-block:: shell-session
+
+       # Intentionally break the Cilium agent by using a non-existent image
+       $ kubectl set image -n kube-system ds/cilium cilium-agent=quay.io/cilium/cilium:non-existent-image
+       
+       # Wait for agent pods to enter ImagePullBackOff state
+       $ kubectl wait --for=condition=Ready=false pod -n kube-system -l k8s-app=cilium --timeout=60s
+       
+       # DNS queries for allowed domains should still work via standalone DNS proxy
+       $ kubectl exec test-pod -- nslookup cilium.io
+
+       # Verify that policy enforcement still works - non-allowed domains are still refused
+       $ kubectl exec test-pod -- nslookup example.com
+       Server:         10.96.0.10
+       Address:        10.96.0.10#53
+
+       ** server can't find example.com: REFUSED
+
+    Both queries demonstrate that the standalone DNS proxy continues to enforce DNS policies
+    independently, allowing ``cilium.io`` and refusing ``example.com``, even when the agent is unavailable.
+
+11. **Restore Cilium Agent**
+
+    Restore the Cilium agent to normal operation:
+
+    .. code-block:: shell-session
+
+       $ kubectl rollout undo ds/cilium -n kube-system
+
+For more information on DNS policies, see :ref:`DNS based`.
+
+Limitations
+===========
+
+The Standalone DNS Proxy alpha release has the following known limitations:
+
+* The standalone DNS proxy uses the agent's DNS configuration (``dnsProxy.proxyPort`` 
+  and ``dnsProxy.enableDnsCompression``). The ``dnsProxy.proxyPort`` must be explicitly 
```

Comment:
I understand why the port is needed (because both sockets need to bind to the same one), but why does one have to explicitly set compression? Why can't we use the same defaults in both components?